File: frontend/src/types/store.ts
// Store Types
export enum StoreStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  SUSPENDED = 'SUSPENDED',
}

export interface Store {
  id: number;
  name: string;
  subscriptionEndDate: string | null;
  status: StoreStatus;
  createdAt: string;
  subscriptionPlanId: number | null;
}

// Subscription Plan Types
export interface SubscriptionPlan {
  id: number;
  name: string;
  price: number;
  description: string | null;
  maxOrdersPerMonth: number;
}

export interface StoreWithPlan extends Store {
  subscriptionPlan: SubscriptionPlan | null;
}

File: backend/admin/src/main/java/com/example/demo/domain/entity/Store.java
package com.example.demo.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "stores")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Store {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    // Liên kết với Gói dịch vụ
    @ManyToOne
    @JoinColumn(name = "subscription_plan_id")
    private SubscriptionPlan subscriptionPlan;

    private LocalDateTime subscriptionEndDate;

    @Column(length = 20)
    private String status; // ACTIVE, LOCKED

    @OneToMany(mappedBy = "store")
    private List<User> users; // Một cửa hàng có nhiều user (chủ, nhân viên)

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) status = "ACTIVE";
    }
}